src/queue.rs

```diff
@@ -130,6 +130,7 @@ pub struct DescriptorChain<M: GuestAddressSpace> {
 
     /// The current descriptor
     desc: Descriptor,
+    curr_indirect: Option<Box<DescriptorChain<M>>>,
```

Comment:
Why does this need `Box`?

Reply:
Otherwise it will fail due to recursive type definition.
```
DescriptorChain {
    curr_indirect: Option<Box<DescriptorChain<M>>>,
}
```

Reply:
Doh, of course.